micro-rdk-server/esp32/main.rs

```diff
@@ -101,30 +101,39 @@ mod esp32 {
                     .into(),
                 )
                 .expect("Failed to store robot credentials to NVS");
-
-            serve_web(None, repr, max_connection, storage);
         }
-        #[cfg(not(has_robot_config))]
-        {
-            use micro_rdk::common::provisioning::server::ProvisioningInfo;
+
+        let max_connections = unsafe {
+            if !g_spiram_ok {
+                log::info!("spiram not initialized disabling cache feature of the wifi driver");
+                g_wifi_feature_caps &= !(CONFIG_FEATURE_CACHE_TX_BUF_BIT as u64);
+                1
+            } else {
+                3
+            }
+        };
+
+        let info = if cfg!(feature = "provisioning") {
             let mut info = ProvisioningInfo::default();
             info.set_manufacturer("viam".to_owned());
             info.set_model("test-esp32".to_owned());
-            let storage = NVSStorage::new("nvs").unwrap();
-            serve_web(Some(info), repr, max_connection, storage);
+            Some(info)
+        } else {
+            None
+        };
+
+        if info.is_none() && !(storage.has_wifi_credentials() && storage.has_robot_credentials()) {
```

Comment:
I wonder a bit if this bit of logic doesn't belong in the entry files rather than the mains. But, I'm just thinking aloud. If I feel strongly about it I'll move it later.

Reply:
I agree since `serve_web` will now have all the info it needs to determine this undesireable state.
Will add a ticket and TODO comment before merging

Reply:
[RSDK-8445](https://viam.atlassian.net/browse/RSDK-8445)